gslib/cloud_api.py

```diff
@@ -120,15 +120,16 @@ def SetBucketIamPolicy(self, bucket_name, policy, provider=None):
     """
     raise NotImplementedError('SetBucketIamPolicy must be overloaded')
 
-  def SignUrl(self, method, duration, path, logger, region, signed_headers,
-              string_to_sign_debug):
+  def SignUrl(self, method, duration, path, generation, logger, region,
+              signed_headers, string_to_sign_debug):
     """Sign a url using service account's system managed private key.
 
     Args:
       method: The HTTP method to be used with the signed URL.
       duration: timedelta for which the constructed signed URL should be valid.
       path: String path to the bucket or object for signing, in the form
           'bucket' or 'bucket/object'.
+      generation: If not None, specifies a version of an object for signing.
```

Comment:
Maybe this should be the last argument with a default of None?

Reply:
I kind of like grouping it with path here, and giving it a default here doesn't change the code much since all the `generation=None` lines in the tests were in calls to a private helper function. I gave generation a default there, but left everything else as-is.